conductor/isolate/isolate.cpp

```diff
@@ -0,0 +1,257 @@
+
+        m_fd = ::open(m_stdout_pipe_path.c_str(), O_RDONLY);
+        if(m_fd == -1){
+            throw std::system_error(errno, std::system_category(), "unable to open an output named pipe");
+        }
+
+        // if(::pipe(pipes.data()) != 0) {
+        //     throw std::system_error(errno, std::system_category(), "unable to create an output pipe");
+        // }
+
+        // if(::close(pipes[1])){
+        //     throw std::system_error(errno, std::system_category(), "unable to close writable side of ::pipe() result");
+        // }
+
+    }
+
+    virtual
+    int
+    stdout() const {
+        return m_fd;
+    }
+
+private:
+    std::shared_ptr<rpc_isolate_t> m_parent;
+    bool m_terminated;
+    uint64_t m_container_id;
+    std::string m_stdout_pipe_path;
+    int m_fd;
+};
+
+} // namespace
+
+rpc_isolate_t::rpc_isolate_t(context_t& context,  asio::io_service& io_context, const std::string& name, const dynamic_t& args):
+    category_type(context, io_context, name, args),
+    m_log(context.log(name, {{"isolate", "rpc"}})),
+    m_do_pool(false),
+    m_context(context),
+    m_args(args)
+{
+    try {
+        const auto& config = args.as_object();
+        m_runtime_path = config.at("runtime-path", "/run/cocaine").as_string();
+
+        if(config.count("registry") > 0) {
+            m_do_pool = true;
+
+            // <default> means default docker's registry (the one owned by dotCloud). User must specify it explicitly.
+            // I think that if there is no registry in the config then the plugin must not pull images from foreign registry by default.
+            if(config["registry"].as_string() != "<default>") {
+                m_image += config["registry"].as_string() + "/";
+            }
+        } else {
+            COCAINE_LOG_WARNING(m_log, "docker registry is not specified - only local images will be used");
+        }
+
+        if(config.count("repository") > 0) {
+            m_image += config["repository"].as_string() + "/";
+        }
+        m_image += name;
+        m_tag = ""; // empty for now
+
+    } catch(const std::exception& e) {
+        throw cocaine::error_t("%s", e.what());
+    }
+}
+
+rpc_isolate_t::~rpc_isolate_t() {
+    COCAINE_LOG_DEBUG(m_log, "isolate is being destroyed");
+}
+
+std::unique_ptr<api::cancellation_t>
+rpc_isolate_t::async_spool(std::function<void(const std::error_code&)> handler) {
+    if(m_do_pool) {
+        COCAINE_LOG_INFO(m_log, "isolate::spool(%s)", m_image);
+
+        auto actor = m_context.locate("conductor");
+
+        if (!actor) {
+            COCAINE_LOG_ERROR(m_log, "unable to spool container: conductor is not available");
+            throw cocaine::error_t("unable to spool container: conductor is not available");
+        }
+
+        auto conductor = dynamic_cast<const service::conductor_t*>(&actor.get().prototype());
+        
+        const_cast<service::conductor_t*>(conductor)->spool(m_image, [&] (const std::error_code& ec) {
```

Comment:
@3Hren @antmat @noxiouz @kobolog is it OK to cast actor->conductor_t like this? If not, how should I communicate with service from runtime?